eth/watchers/senderwatcher_test.go

```diff
@@ -0,0 +1,595 @@
+	time.Sleep(2 * time.Millisecond)
+	watcher.sink <- []*blockwatch.Event{blockEvent}
+	time.Sleep(2 * time.Millisecond)
+	info, err = sw.GetSenderInfo(stubSender)
+	assert.Nil(err)
+	assert.Zero(info.Deposit.Cmp(big.NewInt(0)))
+	assert.Zero(info.Reserve.Cmp(big.NewInt(0)))
+
+	// If we don't care about the address, don't handle the event
+	s := pm.RandAddress()
+	sender := ethcommon.LeftPadBytes(s.Bytes(), 32)
+	var senderTopic ethcommon.Hash
+	copy(senderTopic[:], sender[:])
+	newWithdrawalEvent.Topics[1] = senderTopic
+	watcher.sink <- []*blockwatch.Event{blockEvent}
+	time.Sleep(2 * time.Millisecond)
+	_, ok = sw.senders[s]
+	assert.False(ok)
+}
+
+func TestWinningTicketTransferEvent(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+	deposit, _ := new(big.Int).SetString("10000000000000000000", 10)
+	reserve, _ := new(big.Int).SetString("5000000000000000000", 10)
+	startDeposit, _ := new(big.Int).SetString("10000000000000000000", 10)
+	startReserve, _ := new(big.Int).SetString("5000000000000000000", 10)
+	senderInfo := &pm.SenderInfo{
+		Deposit: deposit,
+		Reserve: reserve,
+	}
+	lpEth := &eth.StubClient{
+		SenderInfo: senderInfo,
+	}
+	watcher := &stubBlockWatcher{}
+	sw, err := NewSenderWatcher(stubTicketBrokerAddr, watcher, lpEth)
+	require.Nil(err)
+
+	header := defaultMiniHeader()
+	newWinningTicketEvent := newStubWinningTicketLog()
+	header.Logs = append(header.Logs, newWinningTicketEvent)
+
+	blockEvent := &blockwatch.Event{
+		Type:        blockwatch.Added,
+		BlockHeader: header,
+	}
+
+	go sw.Watch()
+	defer sw.Stop()
+	time.Sleep(2 * time.Millisecond)
+
+	// If map entry is nil don't handle event
+	watcher.sink <- []*blockwatch.Event{blockEvent}
+	time.Sleep(2 * time.Millisecond)
+	info, ok := sw.senders[stubSender]
+	assert.False(ok)
+
+	// if block removed, make RPC call
+	blockEvent.Type = blockwatch.Removed
+	watcher.sink <- []*blockwatch.Event{blockEvent}
+	time.Sleep(2 * time.Millisecond)
+	info, err = sw.GetSenderInfo(stubSender)
+	assert.Nil(err)
+	assert.Equal(info, lpEth.SenderInfo)
+
+	// If map entry exists parse event log
+	// ticket amount is 200 gwei
+	// init the map entry
+	blockEvent.Type = blockwatch.Added
+	info, err = sw.GetSenderInfo(stubSender)
+	require.Nil(err)
+	require.Equal(info.Deposit, deposit)
+	faceValue := big.NewInt(200000000000)
+	watcher.sink <- []*blockwatch.Event{blockEvent}
+	time.Sleep(2 * time.Millisecond)
+	info, err = sw.GetSenderInfo(stubSender)
+	assert.Nil(err)
+	assert.Zero(startDeposit.Sub(startDeposit, faceValue).Cmp(info.Deposit))
+	assert.Zero(startReserve.Cmp(info.Reserve))
+
+	// Test facevalue > deposit
+	senderInfo.Deposit = big.NewInt(100000000000)
+	info, err = sw.GetSenderInfo(stubSender)
+	watcher.sink <- []*blockwatch.Event{blockEvent}
+	time.Sleep(2 * time.Millisecond)
+	info, err = sw.GetSenderInfo(stubSender)
+	assert.Nil(err)
+	assert.Zero(info.Deposit.Int64())
+	diff := new(big.Int).Abs(new(big.Int).Sub(big.NewInt(100000000000), faceValue))
```

Comment:
We don't need to take the absolute value here and could instead just write: `new(big.Int).Sub(faceValue, big.NewInt(100000000000))`

Reply:
fixed in 4530eff